src/main/java/com/epam/reportportal/elastic/SimpleElasticSearchClient.java

```diff
@@ -24,65 +24,120 @@
+
+    this.host = host;
+  }
+
+  public void save(TreeMap<Long, List<LogMessage>> logMessageMap) {
+    if (CollectionUtils.isEmpty(logMessageMap)) {
+      return;
+    }
+    String create = "{\"create\":{ }}\n";
+
+    logMessageMap.descendingMap().forEach((projectId, logMessageList) -> {
+      String indexName = "logs-reportportal-" + logMessageList.get(0).getProjectId();
+      StringBuilder jsonBodyBuilder = new StringBuilder();
+      for (LogMessage logMessage : logMessageList) {
+        jsonBodyBuilder.append(create).append(convertToJson(logMessage)).append("\n");
+      }
+      restTemplate.put(host + "/" + indexName + "/_bulk?refresh",
+          getStringHttpEntity(jsonBodyBuilder.toString())
+      );
+    });
+  }
+
+  public Optional<LogMessage> getFirstLogFromElasticSearch() {
+
+    SearchResponse searchResponse =
+        restTemplate.postForObject(host + "/_search", getStringHttpEntity(SEARCH_QUERY_JSON),
+            SearchResponse.class
+        );
+
+    if (searchResponse != null && searchResponse.getHits() != null
+        && searchResponse.getHits().getHits() != null && !searchResponse.getHits().getHits()
+        .isEmpty()) {
+      return Optional.of(searchResponse.getHits().getHits().get(0).getSource());
+    }
+
+    return Optional.empty();
+  }
+
+  public Optional<LogMessage> getLogFromLaunch(Long launchId, Long projectId) {
+    String indexName = "logs-reportportal-" + projectId;
+    String query = SEARCH_BY_LAUNCH_ID_JSON.replace("?", launchId.toString());
+    SearchResponse searchResponse =
+        restTemplate.postForObject(host + "/" + indexName + "/_search?size=1",
+            getStringHttpEntity(query), SearchResponse.class
+        );
+
+    if (searchResponse != null && !CollectionUtils.isEmpty(searchResponse.getHits().getHits())) {
+      return Optional.of(searchResponse.getHits().getHits().get(0).getSource());
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  public void updateLogsLaunchIdByTestItemId(List<Long> testItemIds, Long launchId,
+      Long projectId) {
+    String indexName = "logs-reportportal-" + projectId;
+    if (CollectionUtils.isEmpty(testItemIds)) {
+      return;
+    }
+
+    String logUpdateBody = getUpdateLaunchIdJson(testItemIds, launchId).toString();
+
+    restTemplate.postForEntity(host + "/" + indexName + "/_update_by_query",
+        getStringHttpEntity(logUpdateBody), String.class
+    );
+  }
+
+  private JSONObject convertToJson(LogMessage logMessage) {
+    JSONObject personJsonObject = new JSONObject();
+    personJsonObject.put("id", logMessage.getId());
+    personJsonObject.put("message", logMessage.getLogMessage());
+    personJsonObject.put("itemId", logMessage.getItemId());
+    personJsonObject.put("@timestamp", logMessage.getLogTime());
+    personJsonObject.put("launchId", logMessage.getLaunchId());
+
+    return personJsonObject;
+  }
+
+  private JSONObject getUpdateLaunchIdJson(List<Long> testItemIds, Long newLaunchId) {
```

Comment:
seems you can rewrite method (lines 120-133)

```
    return new JSONObject()
        .put("script", "ctx._source.launchId = " + newLaunchId)
        .put("query", new JSONObject()
            .put("terms", new JSONObject()
                .put("itemId", new JSONArray(testItemIds))));
```

Reply:
Rewritten